src/main/webapp/app/course/competencies/competency-management/competency-management-table.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output, inject } from '@angular/core';
+import { CompetencyService } from 'app/course/competencies/competency.service';
+import { AlertService } from 'app/core/util/alert.service';
+import { CompetencyRelation, CompetencyRelationDTO, CompetencyWithTailRelationDTO, CourseCompetency, dtoToCompetencyRelation, getIcon } from 'app/entities/competency.model';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { filter, map } from 'rxjs/operators';
+import { onError } from 'app/shared/util/global.utils';
+import { Subject } from 'rxjs';
+import { faFileImport, faPencilAlt, faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ImportAllCompetenciesComponent, ImportAllFromCourseResult } from 'app/course/competencies/competency-management/import-all-competencies.component';
+import { PrerequisiteService } from 'app/course/competencies/prerequisite.service';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisMarkdownModule } from 'app/shared/markdown.module';
+
+@Component({
+    selector: 'jhi-competency-management-table',
+    templateUrl: './competency-management-table.component.html',
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisMarkdownModule],
+})
+export class CompetencyManagementTableComponent implements OnInit, OnDestroy {
+    @Input() courseId: number;
+    @Input() courseCompetencies: CourseCompetency[];
+    @Input() relations: CompetencyRelation[];
+    @Input() competencyType: 'competency' | 'prerequisite';
```

Comment:
Can we not use the CompetencyType enum here? (From the CourseCompetency interface)?